test/powershell/Modules/Microsoft.PowerShell.Utility/Group-Object.Tests.ps1

```diff
@@ -173,6 +173,20 @@ Describe "Check 'Culture' parameter in order object cmdlets (Group-Object, Sort-
             $testCulture = "1049"
         }
 
-        {$testObject | Group-Object -Culture $testCulture } | Should -Not -Throw
+        { $testObject | Group-Object -Culture $testCulture } | Should -Not -Throw
+    }
+
+    It 'should not throw a key duplication error with -CaseSensitive -AsHashtable' {
+        $capitonyms = @(
+            [PSCustomObject]@{
+                Capitonym = 'Bill'
+            }
+            [PSCustomObject]@{
+                Capitonym = 'bill'
+            }
+        )
+
+        [hashtable] $Result = $capitonyms | Group-Object -Property Capitonym -AsHashTable -CaseSensitive
```

Comment:
If this gets regressed in the future, I think this will give a better test error message

```suggestion
        $Result = $capitonyms | Group-Object -Property Capitonym -AsHashTable -CaseSensitive
        $Result | Should -BeOfType [HashTable]
```

Reply:
Good catch! 🙂